### PR TITLE
If a user provides bad credentials, throw an error.

### LINKED ISSUE #67: Fail loudly with an invalid token

### PR DESCRIPTION
**Pull request by parkr**
This fixes #67. If you give me `ACCESS_TOKEN=1234`, you'll receive a loud error when you run `jekyll serve` that it's invalid. Providing either no token or a valid token will solve the error.

/cc @jekyll/gh-pages for review.

### CODE REVIEW

File: lib/jekyll-github-metadata/client.rb [hunk 1]
```diff
@@ -71,13 +72,25 @@ def inspect
         "#<#{self.class.name} @client=#{client_inspect}>"
       end
 
+      def authenticated?
+        !@client.access_token.nil? && !@client.access_token.empty?
```
Comment (benbalter) [inline lib/jekyll-github-metadata/client.rb]: Is this better as `to_s.empty?`?

File: lib/jekyll-github-metadata/client.rb [hunk 2]
```diff
@@ -71,13 +72,25 @@ def inspect
...
+      end
+
+      # Raise an error if credentials are provided and they cause a 401.
+      def check_credentials!
+        return unless authenticated?
+        @client.user
```
Comment (benbalter) [inline lib/jekyll-github-metadata/client.rb]: Why make an extra call to check credentials, when we can just rescue if the call we need fails due to bad auth?

File: lib/jekyll-github-metadata/ghp_metadata_generator.rb [hunk 3]
```diff
@@ -8,6 +8,9 @@ class GHPMetadataGenerator < Jekyll::Generator
       def generate(site)
         Jekyll::GitHubMetadata.log :debug, "Initializing..."
 
+        # Fail loudly if credentials are provided but aren't valid.
+        Jekyll::GitHubMetadata.client.check_credentials!
```
Comment (benbalter) [inline lib/jekyll-github-metadata/ghp_metadata_generator.rb]: Won't this be a change in behavior? Right now, I can have an invalid credential, but if I don't use any `site.github` metadata, the drop isn't called, and thus it's okay, right? I don't know that we need to go out of our way to fail a user's build if they're not relying on the plugin for their site to build.

**Comment by parkr** [comment]
@benbalter Ready for another round of review!

File: lib/jekyll-github-metadata/client.rb [hunk 4]
```diff
@@ -15,6 +16,7 @@ class Client
         releases
         list_repos
         organization_public_members
+        user
```
Comment (benbalter) [inline lib/jekyll-github-metadata/client.rb]: Do we need this still?

**Comment by benbalter** [comment]
@parkr my bad. One question, which I think is left over from the previous iteration. 

Can this go in a point release, or does it need its own minor release?

**Comment by parkr** [comment]
@benbalter OK. Ready for another review.

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Pull request merged**